### PR TITLE
Update to ensure compatability with Changes to Oldenborg and ue5env

### PR DESCRIPTION
**Pull request by Simon-Heck**

### CODE REVIEW

File: box_unreal_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+from ue5env import UE5EnvWrapper
+import ue5env
+import sys
+from boxnavigator import Action, BoxNavigatorBase
+from math import degrees, radians
+from box import Box, Pt
+
+sys.path.append(
+    "C:/Users/simon/OneDrive/Documents/ArcsLab/ArcLabPrograms/OldenborgTraining/DataCollection"
+)
```
Comment (anthonyjclark) [inline box_unreal_wrapper.py]: This should not be hard-coded.

File: box_unreal_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+from UE5_Data_collector import UE5_data_collection
```
Comment (anthonyjclark) [inline box_unreal_wrapper.py]: Where is this defined? Also, use `UE5DataCollector` instead.

File: box_unreal_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+class NavigatorUnrealWrapper:
+
+    def __init__(
+        self, navigator: BoxNavigatorBase, port: int = 8500, collect_data: bool = False
+    ) -> None:
+        self.experiment_name = "Test"
```
Comment (anthonyjclark) [inline box_unreal_wrapper.py]: What is this?

File: box_unreal_wrapper.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        self.ue5 = UE5EnvWrapper(port)
+        self.navigator = navigator
+        self.collect_data = collect_data
+        dataset_path = "C:/Users/simon/OneDrive/Documents/ArcsLab/ArcLabPrograms/UE5Images/small_test_folder"
```
Comment (anthonyjclark) [inline box_unreal_wrapper.py]: Probably shouldn't be hard-coded.

File: box_unreal_wrapper.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        path_to_unreal_project_image = "C:/Users/simon/OneDrive/Documents/Unreal Projects/OldenborgUE/Saved/Screenshots/WindowsEditor/highres.png"
```
Comment (anthonyjclark) [inline box_unreal_wrapper.py]: Same.

File: box_unreal_wrapper.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        self.UE5_data_collector = UE5_data_collection(
+            self.ue5, self.experiment_name, dataset_path, path_to_unreal_project_image
+        )
+        self.syncUnrealPositionToBox()
+        self.syncRotation()
+
+    def syncUnrealPositionToBox(self) -> None:
```
Comment (anthonyjclark) [inline box_unreal_wrapper.py]: Use `snake_case` for method names. See: https://pep8.org/

File: box_unreal_wrapper.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+            self.UE5_data_collector.collectData(correct_action)
+
+        if action_taken == Action.FORWARD:
+            self.ue5.forward(self.navigator.translation_increment)
+        elif action_taken == Action.BACKWARD:
+            self.ue5.back(self.navigator.translation_increment)
+        elif action_taken == Action.ROTATE_LEFT:
+            self.syncRotation()
+        elif action_taken == Action.ROTATE_RIGHT:
+            self.syncRotation()
```
Comment (anthonyjclark) [inline box_unreal_wrapper.py]: Why are some `self.ue5.forward` and some `self.syncRotation`. Can these be made more consistent?

File: boxenv.py [hunk 8]
```diff
@@ -56,8 +57,7 @@ def display(self, ax: plt.Axes) -> None:
                     box.origin, box.width, box.height, box.angle_degrees, fill=None
                 )
             )
-
-        ax.set_xlim(self.xlim)
+        ax.set_xlim(self.xlim[::-1])
```
Comment (anthonyjclark) [inline boxenv.py:60]: What's with `::-1`?

File: boxnavigator.py [hunk 9]
```diff
@@ -42,29 +42,32 @@ class BoxNavigatorBase:
     """
 
     def __init__(
-        self, position: Pt, rotation: float, env: BoxEnv, out_of_bounds: bool
+        self, position: Pt, rotation: float, env: BoxEnv, out_of_bounds: bool = False
```
Comment (anthonyjclark) [inline boxnavigator.py:45]: `out_of_bounds` missing from docstring.

File: boxnavigator.py [hunk 10]
```diff
@@ -42,29 +42,32 @@ class BoxNavigatorBase:
...
         """
         self.env = env
         self.position = position
         self.rotation = rotation
+
         self.allow_out_of_bounds = out_of_bounds
 
         self.target = self.env.boxes[0].target
+        # experimental:
+        self.isOutOfBounds = False
```
Comment (anthonyjclark) [inline boxnavigator.py]: Snake case.

File: boxnavigator.py [hunk 11]
```diff
@@ -42,29 +42,32 @@ class BoxNavigatorBase:
...
 
         # TODO: change from wedge to field-of-view?
         self.half_target_wedge = radians(5)
 
         # How much a navigator should translate or rotate in a given step
         # of the simulation. These are fairly arbitrary.
-        self.distance_threshold = 15
-        self.translation_increment = 10
+        self.distance_threshold = 50
+        self.translation_increment = 50
```
Comment (anthonyjclark) [inline boxnavigator.py:71]: Can you give an idea of the units for `50`?

File: boxnavigator.py [hunk 12]
```diff
@@ -238,7 +233,7 @@ def __init__(
         ]
 
         # TODO: make this a parameter
-        self.chance_of_random_action = 0.5
+        self.chance_of_random_action = 0.25
```
Comment (anthonyjclark) [inline boxnavigator.py]: Should make it a parameter.

File: boxsim.py [hunk 13]
```diff
@@ -1,28 +1,46 @@
...
-    # Box(Pt(-365, 600), Pt(-450, 600), Pt(-190, 240), Pt(-700, 240)),
+    Box(Pt(-185, 1060), Pt(420, 1060), Pt(420, -350), Pt(10, 420)),
+    Box(Pt(-1110, 590), Pt(420, 590), Pt(420, 245), Pt(-770, 420)),
+    Box(Pt(-855, -100), Pt(-855, 590), Pt(-700, 590), Pt(-780, 0)),
+    Box(Pt(-700, 65), Pt(-700, -100), Pt(-4690, -100), Pt(-4500, -32)),
+    Box(Pt(-4690, 65), Pt(-4415, 65), Pt(-4415, -2400), Pt(-4570, -433)),
+    Box(Pt(-4415, -295), Pt(-4415, -640), Pt(-5755, -640), Pt(-5590, -450)),
+    Box(Pt(-5530, -2400), Pt(-5755, -2400), Pt(-5755, 2845), Pt(-5655, 1983)),
+    # Box(Pt(-5755, 1680), Pt(-5755, -2350), Pt(-4600, 2350), Pt(-5000, 1983)),
 ]
```
Comment (anthonyjclark) [inline boxsim.py:34]: Would be great to include an image.

File: boxsim.py [hunk 14]
```diff
@@ -35,16 +53,20 @@ def simulate():
     else:
         raise ValueError("Invalid argument error (check code for options).")
 
+    if args.ue:
+        agent = NavigatorUnrealWrapper(agent, port=8500, collect_data=args.collect)
+        agent.reset()
+    # else:
+    #     navUnrealWrapper = None
```
Comment (anthonyjclark) [inline boxsim.py]: Delete.

**Review by anthonyjclark — CHANGES_REQUESTED** [review]

File: box_unreal_wrapper.py (hunk 1 above)
Comment (Simon-Heck) [reply box_unreal_wrapper.py]: Unsure how to set up the program where you don't have to hard code in the paths to the other repositories

**Pull request merged**